protocols/kad/src/behaviour.rs

```diff
@@ -753,6 +971,34 @@ where
                             self.queued_events.push(event);
                         }
                     },
+                    QueryInfoInner::GetValue { key: _, results } => {
+                        let result = match results.first() {
```

Comment:
```suggestion
                        // TODO: Return all results. The query should finish as soon as enough results have been collected.
                        let result = match results.into_iter().next() {
```
With `into_iter().next()` there is no need to clone the record one line below, as it is moved.